### PR TITLE
Mark pending submissions after freeze time in scoreboard with different color

### LINKED ISSUE #82: Mark pending submissions after freeze time in scoreboard with different color
Only applies to ICPC contests with frozen scoreboard module enabled.

Background: blue
Score/penalty: `?`

### PR DESCRIPTION
**Pull request by aguss787**
Resolve #82

### CODE REVIEW

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 1]
```diff
@@ -44,22 +44,36 @@ public String computeToString(
...
 
         List<String> problemJids = scoreboardState.getProblemJids();
         Set<String> problemJidsSet = ImmutableSet.copyOf(problemJids);
         Set<String> contestantJids = scoreboardState.getContestantJids();
 
         Map<String, List<Submission>> submissionsMap = new HashMap<>();
-        submissions.forEach(s -> {
-            submissionsMap.putIfAbsent(s.getUserJid(), new ArrayList<>());
-            submissionsMap.get(s.getUserJid()).add(s);
+        Map<String, List<Submission>> pendingSubmissionsMap = new HashMap<>();
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]: `frozenSubmissionsMap`, to be consistent with the naming?

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 2]
```diff
@@ -82,16 +96,10 @@ public String computeToString(
...
                 String problemJid = submission.getProblemJid();
 
-                if (!contestantJids.contains(submission.getUserJid())) {
-                    continue;
-                }
-                if (!problemJidsSet.contains(problemJid)) {
-                    continue;
-                }
-                if (!problemStateMap.get(problemJid).equals(IcpcScoreboardProblemState.NOT_ACCEPTED)) {
+                if (!submission.getLatestGrading().isPresent()) {
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]: Put this also in filtered submissions predicate above?

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 3]
```diff
@@ -119,15 +127,24 @@ public String computeToString(
                 }
             }
 
+            List<Submission> pendingSubmissions = pendingSubmissionsMap.get(contestantJid);
+            for (Submission submission : pendingSubmissions) {
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]: Always inline things like this in the for itself, for simple expressions

```
for (Submission submission : pendingSubmissionsMap.get(contestantJid))
```

File: judgels-frontends/raphael/src/routes/uriel/contests/single/scoreboard/IcpcScoreboardTable/IcpcScoreboardTable.test.tsx [hunk 4]
```diff
@@ -103,13 +103,15 @@ describe('IcpcScoreboardTable', () => {
...
+          ? 'D '
+          : td === 'accepted' ? 'G ' : td === 'not-accepted' ? 'R ' : td === 'frozen' ? 'F ' : 'X ';
     const mapCell = td => getColor(td.prop('className')) + td.find('strong').text() + '/' + td.find('small').text();
     const mapRow = tr => [2, 3, 4, 5].map(x => tr.childAt(x)).map(mapCell);
     const points = wrapper
       .find('tbody')
       .children()
       .map(mapRow);
-    expect(points).toEqual([['3/66', 'G 1/3', 'D 3/14', 'D 1/9'], ['1/17', 'R 1/-', 'G 1/17', 'X -/-']]);
+    expect(points).toEqual([['3/66', 'G 1/3', 'D 3/14', 'D 1/9'], ['1/17', 'R 1/-', 'G 1/17', 'F ?/?']]);
```
Comment (fushar) [inline judgels-frontends/raphael/src/routes/uriel/contests/single/scoreboard/IcpcScoreboardTable/IcpcScoreboardTable.test.tsx]: You're deleting a valid test case here (`X` color)

**Review by fushar — CHANGES_REQUESTED** [review]

**Comment by fushar** [comment]
IMPORTANT! This mark should only apply to FROZEN scoreboard. The OFFICIAL scoreboard should not be affected.

**Comment by aguss787** [comment]
> 
> 
> IMPORTANT! This mark should only apply to FROZEN scoreboard. The OFFICIAL scoreboard should not be affected.

Yup, it is

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 5]
```diff
@@ -82,16 +97,7 @@ public String computeToString(
...
-                    continue;
-                }
-                if (!problemJidsSet.contains(problemJid)) {
-                    continue;
-                }
-                if (!problemStateMap.get(problemJid).equals(IcpcScoreboardProblemState.NOT_ACCEPTED)) {
-                    continue;
-                }
-                if (!submission.getLatestGrading().isPresent()) {
+                if (isAccepted(problemStateMap.get(submission.getProblemJid()))) {
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]:
```suggestion
                if (isAccepted(problemStateMap.get(problemJid))) {
```

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 6]
```diff
@@ -119,15 +125,23 @@ public String computeToString(
                 }
             }
 
+            for (Submission submission : frozenSubmissionsMap.get(contestantJid)) {
+                if (isAccepted(problemStateMap.get(submission.getProblemJid()))) {
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]:
```suggestion
                if (isAccepted(problemStateMap.get(problemJid))) {
```

File: judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java [hunk 7]
```diff
@@ -119,15 +125,23 @@ public String computeToString(
...
+                    continue;
+                }
+
+                problemStateMap.put(submission.getProblemJid(), IcpcScoreboardProblemState.FROZEN);
```
Comment (fushar) [inline judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java]:
```suggestion
                problemStateMap.put(problemJid, IcpcScoreboardProblemState.FROZEN);
```

**Review by fushar — APPROVED** [review]
Please apply the autosuggestions and then it is good to go

**Pull request merged**